src/tunnel.cc

```diff
@@ -1016,8 +1067,11 @@ void
 TunnelStateData::connectedToPeer(Security::EncryptorAnswer &answer)
 {
     if (ErrorState *error = answer.error.get()) {
-        answer.error.clear(); // sendError() will own the error
-        sendError(error, "TLS peer connection error");
+        saveError(error);
+        answer.error.clear();
+        if (CachePeer *p = server.conn->getPeer())
+            peerConnectFailed(p);
+        server.conn->close();
```

Comment:
If I understand the intent correctly, I would clarify:

```suggestion
        server.conn->close(); // tunnelServerClosed() will retry as needed
```